odoo/tools/config.py

```diff
@@ -142,13 +142,28 @@ def __init__(self, fname=None):
                          help="Listen port for the gevent worker", type="int", metavar="PORT")
         group.add_option("--no-http", dest="http_enable", action="store_false", my_default=True,
                          help="Disable the HTTP and Longpolling services entirely")
+
+        # HTTP: configure werkzeug proxy mode
+        # https://werkzeug.palletsprojects.com/en/0.16.x/middleware/proxy_fix/
         group.add_option("--proxy-mode", dest="proxy_mode", action="store_true", my_default=False,
                          help="Activate reverse proxy WSGI wrappers (headers rewriting) "
                               "Only enable this when running behind a trusted web proxy!")
+        group.add_option("--proxy-x-for", dest="proxy_x_for", type="int", my_default=1,
+                         help="Number of values to trust for X-Forwarded-For.")
+        group.add_option("--proxy-x-proto", dest="proxy_x_proto", type="int", my_default=1,
+                         help="Number of values to trust for X-Forwarded-Proto.")
+        group.add_option("--proxy-x-host", dest="proxy_x_host", type="int", my_default=0,
+                         help="Number of values to trust for X-Forwarded-Host.")
+        group.add_option("--proxy-x-port", dest="proxy_x_port", type="int", my_default=0,
+                         help="Number of values to trust for X-Forwarded-Port.")
+        group.add_option("--proxy-x-prefix", dest="proxy_x_prefix", type="int", my_default=0,
+                         help="Number of values to trust for X-Forwarded-Prefix.")
+
```

Comment:
I didn't really checked the issue here but in any case we prefer to avoid adding more command line arguments.

Reply:
Just god pinged here and didn't notice it was an old pr with an existing discussion, oupsie